src/java.base/share/classes/java/lang/invoke/LambdaFormEditor.java

```diff
@@ -231,6 +229,22 @@ private static TransformKey ofBothArrays(byte kind, int[] b123, byte[] b456) {
                 PACKED_BYTE_MASK = (1 << PACKED_BYTE_SIZE) - 1,
                 PACKED_BYTE_MAX_LENGTH = (STRESS_TEST ? 3 : 64 / PACKED_BYTE_SIZE);
 
+        private static long packedBytes(byte b0, int b1, int[] b23456) {
+            if (b23456.length + 2 > PACKED_BYTE_MAX_LENGTH)
+                return 0;
+            long pb = 0;
+            int bitset = b0 | b1;
+            for (int i = 0; i < b23456.length; i++) {
+                int b = b23456[i] & 0xFF;
+                bitset |= b;
```

Comment:
Looks like `b` is always truncated. I wonder what happens if the ints in this array are larger than a byte (which seems to be possible in e.g. the case of argument positions). Some higher order bits might be dropped, but the resulting `b` might only have the least significant 4 bits set.

I think the untruncated value should be used to compute the bitset? `butset |= b23456[i]`?

Reply:
Maybe not... argument positions should fit in a byte as well. But, maybe there are other problematic cases? Or are the ints guaranteed to fit in a byte?

Reply:
Yes, please. IMHO it makes it clear that that is the assumption of this code.